src/client/java/teammates/client/scripts/DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat.java

```diff
@@ -0,0 +1,166 @@
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.util.Const;
+import teammates.logic.core.FeedbackResponseCommentsLogic;
+import teammates.storage.search.FeedbackResponseCommentSearchDocument;
+import teammates.storage.search.SearchManager;
+
+/**
+ * Script to fix the date format in old {@link FeedbackResponseCommentSearchDocument}s created before V5.93.
+ *
+ * <p>Before V5.93, we used GSON's default JSON serializer, which output the dates in en-US format.
+ * From V5.93 onwards, we have been using our own JSON serializer which writes dates in our prescribed format.
+ * Comment search documents now exist in both formats, which the current codebase has to handle.</p>
+ *
+ * <p>This script migrates all comment search documents to the newer prescribed date format.</p>
+ */
+public class DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat extends RemoteApiClient {
+    /**
+     * Will not perform updates on the datastore if true.
+     */
+    private static final boolean isPreview = true;
+
+    /**
+     * Number of comments/documents to process per cycle.
+     * Maximum documents that can be updated in one request is 200 (limit imposed by GAE Search API).
+     */
+    private static final int batchSize = 200;
+
+    private static final DateFormat oldDateFormat =
+            DateFormat.getDateTimeInstance(DateFormat.DEFAULT, DateFormat.DEFAULT, Locale.US);
+    private static final DateFormat newDateFormat = new SimpleDateFormat(Const.SystemParams.DEFAULT_DATE_TIME_FORMAT);
+
+    private static final Index index = SearchServiceFactory.getSearchService()
+            .getIndex(IndexSpec.newBuilder().setName(Const.SearchIndex.FEEDBACK_RESPONSE_COMMENT));
+    private static final JsonParser jsonParser = new JsonParser();
+
+    public static void main(String[] args) throws IOException {
+        new DataMigrationForFeedbackResponseCommentSearchDocumentDateFormat().doOperationRemotely();
+    }
+
+    @Override
+    protected void doOperation() {
+        List<FeedbackResponseCommentAttributes> commentsToFix = analyzeAndGetAffectedComments();
+        fixDocuments(commentsToFix);
+    }
+
+    private List<FeedbackResponseCommentAttributes> analyzeAndGetAffectedComments() {
+        println("Analyzing comments and their search documents...");
+
+        List<FeedbackResponseCommentAttributes> allComments =
+                FeedbackResponseCommentsLogic.inst().getAllFeedbackResponseComments();
+
+        List<FeedbackResponseCommentAttributes> commentsToFix = new ArrayList<>();
+
+        int numberOfDocuments = 0;
+        int numberOfUnaffectedDocuments = 0;
+        LoopHelper loopHelper = new LoopHelper(batchSize, "comments analyzed.");
+
+        for (FeedbackResponseCommentAttributes comment : allComments) {
+            loopHelper.recordLoop();
+
+            Document document = getDocument(comment);
+            if (document == null) {
```

Comment:
It might be pretty common (resulting in lots of output), and outside the scope of this issue? If we realise there's a huge discrepancy in the reported numbers of comments vs documents on the live server, a new issue and admin script can be written for it.